clang/lib/Sema/SemaType.cpp

```diff
@@ -7963,6 +7967,148 @@ static Attr *getCCTypeAttr(ASTContext &Ctx, ParsedAttr &Attr) {
   llvm_unreachable("unexpected attribute kind!");
 }
 
+ExprResult Sema::ActOnEffectExpression(Expr *CondExpr, FunctionEffectMode &Mode,
+                                       bool RequireConstexpr) {
+  // see checkFunctionConditionAttr, Sema::CheckCXXBooleanCondition
+  if (RequireConstexpr || !CondExpr->isTypeDependent()) {
+    ExprResult E = PerformContextuallyConvertToBool(CondExpr);
+    if (E.isInvalid())
+      return E;
+    CondExpr = E.get();
+    if (RequireConstexpr || !CondExpr->isValueDependent()) {
```

Comment:
I don’t think we usually handle type dependence and value dependence separately like this in Sema. I think just checking if it’s type dependent *or* value dependent at the start of the function and just treating this as dependent if either is true would be simpler.

Reply:
Thanks. I found some examples of other attributes that are handled more simply, as you describe, will work to simplify.